app/models/course_user.rb

```diff
@@ -1,5 +1,32 @@
 class CourseUser < ActiveRecord::Base
+  stampable
   belongs_to :user, inverse_of: :course_users
   belongs_to :course, inverse_of: :course_users
-  enum role: { student: 0, teaching_assistant: 1, manager: 2, owner: 3 }
+
+  enum role: { student: 0, teaching_assistant: 1, manager: 2, owner: 3 },
+       status: { pending: 0, approved: 1 }
```

Comment:
This ought to be a workflow. see geekq/workflow@403a9e44bf49f4d154156d5701e3d67b115ed6da